src/components/issues/details/IssueDetails.js

```diff
@@ -0,0 +1,90 @@
+// @flow
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import styled from 'styled-components';
+
+import Header from './Header';
+import IssueContent from './IssueContent';
+import SideBar from './SideBar';
+import {
+  setCurrentIssue,
+  fetchComments,
+  removeComments
+} from 'actions/issues.actions';
+
+import type { State } from 'types/redux.types';
+import type { Issue, Comments } from '../issues.types';
+
+type OwnProps = {};
+
+type ConnectedProps = {
+  setCurrentIssue: () => void,
+  fetchComments: () => void,
+  currentIssue: Issue,
+  issueComments: Comments,
+  match: Object,
+  params: Object
+};
+
+class IssueDetails extends Component<ConnectedProps & OwnProps> {
+  componentDidMount() {
+    this.props.setCurrentIssue(this.props.match.params.issueId);
+  }
+
+  componentDidUpdate(prevProps) {
+    if (
+      this.props.currentIssue.comments &&
+      this.props.currentIssue !== prevProps.currentIssue
+    ) {
+      this.props.fetchComments(this.props.currentIssue.comments_url);
+    }
+    console.log('props', this.props);
+  }
+
+  componentWillUnmount() {
+    if (this.props.currentIssue.comments) {
```

Comment:
Instead of firing an action every time the component unmount (which half of the time will be unnecessary) it validates that there is indeed comments to remove from state